wiring/inc/spark_wiring_ble.h

```diff
@@ -395,9 +402,13 @@ class BleCharacteristic {
     ssize_t setValue(const uint8_t* buf, size_t len);
     ssize_t setValue(const String& str);
     ssize_t setValue(const char* str);
+    ssize_t setValue(const uint8_t* buf, size_t len, bool ack);
+    ssize_t setValue(const String& str, bool ack);
+    ssize_t setValue(const char* str, bool ack);
 
     template<typename T>
-    ssize_t setValue(T val) {
+    typename std::enable_if<std::is_integral<T>::value, ssize_t>::type
+    setValue(T val) {
         uint8_t buf[BLE_MAX_ATTR_VALUE_PACKET_SIZE];
         size_t len = std::min(sizeof(T), (unsigned)BLE_MAX_ATTR_VALUE_PACKET_SIZE);
```

Comment:
This also doesn't seem necessary for _integral_ types as they are definitely less than ~250 bytes. And if even if they were larger, it makes more sense not to send anything at all, rather that send up partial data. Same in other places like https://github.com/particle-iot/device-os/pull/1901/files#diff-75b7ed256204b5221da93d699fee0429R422